src/main/java/cs361/battleships/models/Board.java

```diff
@@ -40,32 +40,31 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 		else if (ship.getKind().equals("DESTROYER")){
 			placedShip = new Destroyer();
 		}
-		else {
+		else if (ship.getKind().equals("BATTLESHIP")){
 			placedShip = new Battleship();
 		}
+		else {
+			placedShip = new Submarine(submerged);
+		}
 		placedShip.place(y, x, isVertical);
-		if (ships.stream().anyMatch(s -> s.overlaps(placedShip))) {
-			return false;
+		int i;
+		boolean blocked = false;
+		if (ships.stream().anyMatch(s -> s.overlaps(placedShip)) && !(submerged)) {
+			var overlap = ships.stream().filter(s -> s.overlaps(placedShip)).collect(Collectors.toList());
+			for (i = 0; i < overlap.size(); i++ ){
+				var overlapShip = overlap.get(i);
+				if (!(overlapShip.getSubmerged())) {
+					 blocked = true;
+				}
+			}
+			if (blocked) {
+				return false;
+			}
 		}
```

Comment:
Can you elaborate on what this above if statement does?

Reply:
That part looks to if you already have a ship placed on top of a submarine and makes it so you can't place another ship on top of that if you do. It's because it recognizes that there is a submarine there first which allows a ship to be placed on top, but then it need to block any additional ships after the first one.